packages/core/src/createPrepareTransfers.ts

```diff
@@ -151,6 +151,10 @@ export const createPrepareTransfers = (provider?: Provider, now: () => number =
             }
         }
 
+        if (!(transfers instanceof Array)) {
```

Comment:
I'm sorry for not making this clear in the issue, I think that we should remove `.map` completely from line: https://github.com/iotaledger/iota.js/pull/309/files#diff-28f30180563feb8420184c27ede3909aR163

Then we can just rely on validation in line: 
https://github.com/iotaledger/iota.js/pull/309/files#diff-28f30180563feb8420184c27ede3909aR196

However we will have to add a fallback to empty string in line:
https://github.com/iotaledger/iota.js/pull/309/files#diff-28f30180563feb8420184c27ede3909aR215
like this, including `()` which are important:
```TS
message: NULL_HASH_TRYTES + (transfer.message || ''),
```

This is because message and tag are optional and concatenating `undefined` or `null` would give inconsistent message fragments:
```
'hey world!' + undefined
"hey world!undefined"
```
Tag however is fine because default is set here:
https://github.com/iotaledger/iota.js/blob/47f00b925184302f6a4b728f5eeeaf7f7670a4f0/packages/bundle/src/index.ts#L33

Message case is different because we process it directly from [`Transfer`](https://github.com/iotaledger/iota.js/blob/47f00b925184302f6a4b728f5eeeaf7f7670a4f0/packages/types.ts#L37) type where it's declared as optional. Even if we set it to `''` before processing it, TypeScript compiler will complain about possibly undefined values, so this hack might look weird but it's just fine!

Reply:
I'm not entirely sure if I understand your desired changes. If we remove .map and pass transfers to the validator, then x.every will not work on transfers, unless I throw an error in the validator, which would be more generic (`Not an array` instead of `Invalid transfers array`)